src/main/java/net/tropicraft/core/common/entity/underdasea/EntitySeaTurtle.java

```diff
@@ -1,25 +1,580 @@
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import net.tropicraft.core.common.Util;
 import net.tropicraft.core.common.entity.underdasea.atlantoku.EntityTropicraftWaterBase;
+import net.tropicraft.core.common.entity.underdasea.atlantoku.IAmphibian;
+import net.tropicraft.core.registry.BlockRegistry;
+
+public class EntitySeaTurtle extends EntityTropicraftWaterBase implements IAmphibian {
+
+	private static final long EGG_SITE_WAIT_TIME = 450L;
+	private static final long EGG_INTERVAL_MINIMUM = 2000L;
+	private static final int NEST_SITE_SEARCH_ODDS = 100;
+	private static final int MAX_BLOCK_SCAN_RADIUS = 32;
+
+	private static final DataParameter<Boolean> IS_MATURE = EntityDataManager.<Boolean>createKey(EntitySeaTurtle.class,
+			DataSerializers.BOOLEAN);
+
+	public boolean isSeekingLand = false;
+	public BlockPos currentNestSite = null;
+	public BlockPos targetWaterSite = null;
+	public boolean isLandPathing = false;
+	public boolean isSeekingWater = false;
+	public long timeSinceLastEgg = 0L;
+	public long eggSiteCooldown = EGG_SITE_WAIT_TIME;
+	public PathNavigateGround png;
+	public EntityTurtleEgg lastEgg = null;
+
+	public EntitySeaTurtle(World par1World) {
+		super(par1World);
+		setSize(0.9f, 0.4f);
+		this.setSwimSpeeds(1f, 1f, 1f);
+		this.setApproachesPlayers(true);
+		png = new PathNavigateGround(this, par1World);
+		this.stepHeight = 1f;
+		this.setMaxHealth(3);
+	}
+
+	/** Constructor for baby */
+	public EntitySeaTurtle(World world, int age) {
+		this(world);
+		setSize(0.1f, 0.1f);
+	}
+
+	@Override
+	public void entityInit() {
+		super.entityInit();
+
+		this.getDataManager().register(IS_MATURE, false);
+		this.assignRandomTexture();
+	}
+
+	@Override
+	public String[] getTexturePool() {
+		return new String[] { "seaTurtle", "sea_turtle2", "sea_turtle3", "sea_turtle4", "sea_turtle5", "sea_turtle6" };
+	}
+
+	@Override
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		this.setAir(30);
+		if (!this.isInWater() && this.isBeingRidden()) {
+			this.removePassengers();
+		}
+		if (this.getNavigator() == null) {
+			return;
+		}
+
+		float renderSize = 0.3f + (((float) ticksExisted / 2000));
+		if (renderSize > 1f)
+			renderSize = 1f;
+		if (this.ticksExisted % 40 == 1 && !this.isMature()) {
+			setSize(renderSize, renderSize * 0.5f);
+		}
+
+		if (this.isMature()) {
+			renderSize = 1f;
+			if (this.ticksExisted == 1)
+				setSize(0.9f, 0.4f);
+		}
+
+		if (world.isRemote) {
+			return;
+		}
+
+		if (renderSize >= 1f) {
+			setMature();
+			timeSinceLastEgg++;
+		}
+
+		if (this.timeSinceLastEgg > EGG_INTERVAL_MINIMUM && !isSeekingLand && rand.nextInt(NEST_SITE_SEARCH_ODDS) == 0
+				&& world.getWorldTime() > 13000L && world.getWorldTime() < 23000L) {
```

Comment:
This wouldnt trigger after the first night in world, suggestion: (worldTime % 24000)